### PR TITLE
Add initial filter of inflection points

### PR DESCRIPTION
**Pull request by alecbcs**

### CODE REVIEW

File: models/query.go [hunk 1]
```diff
@@ -90,17 +92,26 @@ func GetOrCreateCommit(commitRequest CommitRequest) Commit {
...
 	// Default page size 100
 	pageSize := 100
 	offset := (pageID - 1) * pageSize
 
-	points := []InflectionPoint{}
-
 	// Do a query for all points and fill in associated fields
-	DB.Preload("Tags").Joins("Package").Joins("Commit").Find(&points).Offset(offset).Limit(pageSize)
-	return &points
+	if len(name) > 0 {
```
Comment (vsoch) [inline models/query.go:101]: Fantastic! So you got it working?

File: models/query.go [hunk 2]
```diff
@@ -90,17 +92,26 @@ func GetOrCreateCommit(commitRequest CommitRequest) Commit {
...
+			DB.Preload("Tags").Joins("Package").Joins("Commit").Where(
+				&InflectionPoint{
+					Package: &pkg,
+				},
+			).Find(&points).Offset(offset).Limit(pageSize)
+		}
+	} else {
+		DB.Preload("Tags").Joins("Package").Joins("Commit").Find(&points).Offset(offset).Limit(pageSize)
+	}
+	return points
```
Comment (vsoch) [inline models/query.go:114]: My one question is if we should try to return an empty list when there aren't any points (as opposed to just no response which seems to happen).

File: models/query.go (hunk 1 above)
Comment (alecbcs) [reply models/query.go:101]: I did! You were right you just have to do the query in two steps.

File: models/query.go (hunk 2 above)
Comment (alecbcs) [reply models/query.go:114]: I think right now I get a `null` response but yeah we can totally return an empty list instead to make it more consistent.

Comment (alecbcs) [reply models/query.go:116]: Okay empty results now return a standard empty list if nothing is found.

**Review by vsoch — APPROVED** [review]

**Pull request merged**